stdsimd/arch/detect/os/aarch64.rs

```diff
@@ -0,0 +1,64 @@
+//! Run-time feature detection for Aarch64 on any OS that traps the mrs instruction.
+
+use arch::detect::Feature;
+use arch::detect::cache;
+
+/// Try to read the features from the system registers.
+///
+/// This will cause SIGILL when the current OS is not trapping the mrs instruction.
+pub(crate) fn detect_features() -> cache::Initializer {
+    let mut value = cache::Initializer::default();
+
+    {
+        let mut enable_feature = |f, enable| {
+            if enable {
+                value.set(f as u32);
+            }
+        };
+
+        // ID_AA64ISAR0_EL1 - Instruction Set Attribute Register 0
+        let aa64isar0: u64;
+        unsafe { asm!("mrs $0, S3_0_C0_C6_0" : "=r"(aa64isar0)); }
+
+        let aes = bits_shift(aa64isar0, 7, 4) >= 1;
+        let pmull = bits_shift(aa64isar0, 7, 4) >= 2;
```

Comment:
According to [arm docs](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.100458_0200_00_en/lau1442496521966.html) should this only check for 0x2? 

Additionally, I wonder if this plus `crypto` below means that we should split up the `crypto` feature if AArch64 actually has multiple features here... (not necessary to do in this PR)

Reply:
(see above)

This is a better example actually: this is the same bits of the same register. `>= 1` means AES is supported, `>= 2` means PMULL is supported.

I like the idea of splitting `crypto`. Looks like the idea here was to align to LLVM's target features, but it's okay to e.g. check for AES only when writing an AES function that would use the `crypto` target feature

Reply:
Ok cool, let's move [discussion for `crypto` to an issue](https://github.com/rust-lang-nursery/stdsimd/issues/614)

Reply:
This looks resolved to me (splitting `crypto` is an orthogonal issue).